vault/expiration.go

```diff
@@ -422,6 +454,31 @@ func (m *ExpirationManager) Restore(errorFunc func()) (retErr error) {
 	m.restoreModeLock.Unlock()
 
 	m.logger.Info("lease restore complete")
+
+	if !m.SHA1HashedLeasesCleared {
+		// At this time all the leases are loaded. The restoration of leases would
+		// have also computed if were any leases that had either its ID or
+		// secondary index obfuscated using SHA1 hash instead of SHA2-256 HMAC.
+		// Persist the result of that computation across reboots.
+		expMetadata := &ExpirationMetadata{}
+
+		if atomic.LoadUint32(&m.SHA1HashedLeasesPresent) == 0 {
+			expMetadata.SHA1HashedLeasesCleared = true
+		}
+
+		m.logger.Debug("persisting expiration manager metadata", "sha1_hashed_leases_cleared", expMetadata.SHA1HashedLeasesCleared)
+
+		// Encode and persist the metadata entry
+		metadataEntry, err := logical.StorageEntryJSON(metadataPath, expMetadata)
+		if err != nil {
+			return err
+		}
+		err = m.view.Put(m.quitContext, metadataEntry)
+		if err != nil {
+			return err
+		}
+	}
+
```

Comment:
You don't persist this value into the running expmgr, but there are cases later on the code where you actually use that value to gate behavior. If it gets set true you should probably update the value in the running expmgr...this will necessarily mean that you make it a uint and use an atomic value.

Reply:
Fixed.